lib/card/query/ref_spec.rb

```diff
@@ -16,9 +21,15 @@ def initialize key, val, parent
     end
 
     def to_sql *args
-      dir, type = REFERENCE_DEFINITIONS[ @key.to_sym ]
+      dir, *type = REFERENCE_DEFINITIONS[ @key.to_sym ]
       field1, field2 = REFERENCE_FIELDS[ dir ]
-      cond = type ? ["ref_type='#{type}'"] : []
+      cond = case type.size 
+        when 0 then [] 
+        when 1 then ["ref_type='#{type.first}'"] 
```

Comment:
I can see it either way, but to me this seems like having two limit cases when they really can fold into one.  Small issue in any case.